Attribute Analysis.ipynb

```diff
@@ -0,0 +1,650 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Attribute Analysis\n",
+    "The following notebook can analyze a given attribute with any number of expected values (2+), and will perform metrics 1,2,3,4,11 generalized to any attribute"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instruction\n",
+    "1. Fill in the dataset in section 1.1\n",
+    "\n",
+    "2. Fill in attribute information in 1.2\n",
+    "\n",
+    "3. Run all of the cells\n",
+    "\n",
+    "4. Look at the summary pdf generated AND/OR explore each metric below.\n",
+    "    - Under each Metric there will be a portion of \"Setup\" and then \"Analyses\". Ignore the \"Setup\" unless customization is needed, and in \"Analyses\" results are shown to be interacted with. The number that comes after the M in the title refers to the measurement number when collecting the metrics.\n",
+    "    \n",
+    "## Table of Contents\n",
+    "1. [Initial Setup](#setup) <br/>\n",
+    "    1.1 [Dataset](#dataset) <br/>\n",
+    "    1.2 [Attribute](#attribute)\n",
+    "2. (M1) Metric: [Distance from center, size, attribute label inference](#metric1)<br/>\n",
+    "    2.1 [Setup](#metric1_setup)<br/>\n",
+    "    2.2 [Analyses](#metric1_analyses)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initial Setup\n",
+    "<a id=\"setup\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from __future__ import print_function\n",
+    "import argparse\n",
+    "import datasets\n",
+    "import pickle\n",
+    "import itertools\n",
+    "import torchvision.transforms as transforms\n",
+    "import torch.utils.data as data\n",
+    "import os\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib as mpl\n",
+    "from sklearn.manifold import TSNE\n",
+    "import seaborn as sns\n",
+    "import numpy as np\n",
+    "from scipy import stats\n",
+    "import PIL.Image\n",
+    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
+    "from math import sqrt\n",
+    "import cv2\n",
+    "import matplotlib.patches as patches\n",
+    "from scipy.spatial.distance import squareform\n",
+    "import pycountry\n",
+    "from geonamescache import GeonamesCache\n",
+    "from matplotlib.patches import Polygon\n",
+    "from matplotlib.collections import PatchCollection\n",
+    "from mpl_toolkits.basemap import Basemap\n",
```

Comment:
this line was removed in another recent pr, so can just delete the basemap import